test/docfx.Test/DocfxLanguageServerTestHost.cs

```
@@ -0,0 +1,92 @@
+           Channel.CreateUnbounded<LanguageServerNotification>();
+
+        private Task startUpTask;
+
+        private static readonly string[] s_notificationsToListen = { "window/showMessage" };
+
+        protected ILanguageClient Client { get; private set; }
+
+        public DocfxLanguageServerTestHost()
+            : base(new JsonRpcTestOptions())
+        {
+        }
+
+        public async Task InitializeAsync()
+        {
+            Client = await InitializeClient(x =>
+            {
+                x.WithCapability(new WorkspaceEditCapability()
+                {
+                    DocumentChanges = true,
+                    FailureHandling = FailureHandlingKind.Undo,
+                });
+
+                foreach (var name in s_notificationsToListen)
+                {
+                    x.OnJsonNotification(name, @params =>
+                    {
+                        _notificationsChannel.Writer.TryWrite(new LanguageServerNotification(name, @params));
+                    });
+                }
+            });
+
+            await startUpTask;
+        }
+
+        public void SendNotification(LanguageServerNotification notification)
+        {
+            Client.SendNotification(notification.Method, notification.Params);
+        }
+
+        public async Task<LanguageServerNotification> GetExpectedNotification(string method)
+        {
+            try
+            {
+                using var cts = new CancellationTokenSource(10000);
+                while (true)
+                {
+                    var notification = await _notificationsChannel.Reader.ReadAsync(cts.Token).AsTask();
+                    if (notification.Method.Equals(method, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return notification;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
```

consider don't catch the exception and let the method throw to fail the test, to tell the different between not receiving such notification and received a notification but the params is `null`